### PR TITLE
Fix test for API strategy. Make `decimals` parameter optional and add Readme.

### PR DESCRIPTION
**Pull request by ori-wagmi**
Changes proposed in this pull request:
- Remove the `symbol` and `decimal` parameters from `example.json`. This was probably copied from erc20 strategy but it's not used at all for the api strategy.
- Add a readme to explain how the api strategy is used.

Verified the api strategy passes all tests

### CODE REVIEW

File: src/strategies/api/examples.json [hunk 1]
```diff
@@ -5,8 +5,6 @@
       "name": "api",
       "params": {
         "api": "https://gateway.pinata.cloud/ipfs",
-        "symbol": "LIFE",
-        "decimals": 0,
```
Comment (ChaituVR) [inline src/strategies/api/examples.json:9]: Awesome thanks @ori-wagmi but we still use decimals here, maybe we can make it optional? (default to `0`) https://github.com/snapshot-labs/snapshot-strategies/blob/d0c8ba48bda1e6e77be09f83bc18316b65edcff6/src/strategies/api/index.ts#L44

File: src/strategies/api/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - `https://gateway.pinata.cloud/ipfs/'
+  - 'https://ipfs.io/ipfs/'
+  - 'https://cloudflare-ipfs.com/ipfs/'
+
+```javascript
+1. `param.api`: The first part of the URL (e.g. `https://gateway.pinata.cloud/ipfs/`)
+
+2. `param.strategy`: The IPFS hash
+
+3. `param.additionalParameters`: Any additional parameters you want to include (i.e. `decimals`)
```
Comment (ChaituVR) [inline src/strategies/api/README.md:22]:
```suggestion
3. `param.additionalParameters` (optional): Any additional parameters you want to include (i.e. `decimals`)
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by ori-wagmi** [comment]
@ChaituVR , seems like i've been testing the strategy incorrectly, I was using `--strategy:api` instead of `--strategy=api`.

Now that I'm actually testing the API strategy, it seems to be failing. Am I missing something, or has this been broken for a while?

Here's the URL that it's hitting: https://gateway.pinata.cloud/ipfs/QmQnW3TtpN8WS2YMXtWB1p7DFcVjetfZmMfJvXm5yAZ6QN

Here's the output:
```
c:\code\snapshot-strategies\src\strategies\api>git branch
* master

c:\code\snapshot-strategies\src\strategies\api>git log
commit 7d706f7826a2a69d3d5c9e251cf2cd8c19a60715 (HEAD -> master, origin/master, origin/HEAD)
Author: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Date:   Fri Feb 3 01:07:24 2023 +0530

    fix(deps): update dependency @snapshot-labs/snapshot.js to ^0.4.61 (#1040)

    Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
    
    c:\code\snapshot-strategies\src\strategies\api>npm run test --strategy=api

> @snapshot-labs/strategies@0.1.0 test
> jest -i strategy.test.ts

 FAIL  test/strategy.test.ts (19.59 s)

Test strategy "api" with example index 0
    √ Strategy name should be lowercase and should not contain any special char expect hyphen (2 ms)
    √ Strategy name should be same as in examples.json
    √ Addresses in example should be minimum 3 and maximum 20 (1 ms)
    √ Must use a snapshot block number in the past (229 ms)
    × Strategy should run without any errors (177 ms)
    × Should return an array of object with addresses (2 ms)
    × Should take less than 10 sec. to resolve (1 ms)
    × File examples.json should include at least 1 address with a positive score
    × Returned addresses should be checksum addresses
    × Voting power should not depend on other addresses (86 ms)

Test strategy "api" with example index 0 (latest snapshot)
    × Strategy should run without any errors (128 ms)
    × Should return an array of object with addresses (1 ms)

Test strategy "api" with example index 0 (with 500 addresses)
    ○ skipped Should work with 500 addresses
    ○ skipped Should take less than 20 sec. to resolve with 500 addresses

Other tests with example index 0
    ○ skipped Check schema (if available) is valid with examples.json
    ○ skipped Strategy should work even when strategy symbol is null

Others:
    √ Author in strategy should be a valid github username (297 ms)
    √ Version in strategy should be a valid string
```

File: src/strategies/api/examples.json (hunk 1 above)
Comment (ori-wagmi) [reply src/strategies/api/examples.json:9]: - i continued to removed decimals from example.json
- made code to default to 0
- updated readme to add a "decimals" section

File: src/strategies/api/README.md (hunk 2 above)
Comment (ori-wagmi) [reply src/strategies/api/README.md]: done

**Comment by ori-wagmi** [comment]
@ChaituVR , So I think this strategy's example.json has been broken for a while. I've uploaded my own json to IPFS and updated the example to match it. https://gateway.pinata.cloud/ipfs/QmbmhTivxYuLE5uhNEALoBmvP7Yg9acA2Lkw9V9PqaEmw6

Here's the current experience:
```
c:\code\snapshot\snapshot-strategies\src\strategies\api>git branch
  master
* ori/updateApiStrategy

c:\code\snapshot\snapshot-strategies\src\strategies\api>git log -n 1
commit 432c005de53272699297c1b78de076d733267353 (HEAD -> ori/updateApiStrategy, origin/ori/updateApiStrategy)
Author: ori-wagmi <ori.wagmi@gmail.com>
Date:   Thu Feb 2 14:25:35 2023 -0800

    make decimals optional, update example.json to working IPFS endpoing

c:\code\snapshot\snapshot-strategies\src\strategies\api>npm run test --strategy=api

> @snapshot-labs/strategies@0.1.0 test
> jest -i strategy.test.ts

  console.log
    [
      {
        '0xeD2bcC3104Da5F5f8fA988D6e9fAFd74Ae62f319': 184000000000000000000,
        '0x3c4B8C52Ed4c29eE402D9c91FfAe1Db2BAdd228D': 7.469258545106344e+24,
        '0xd649bACfF66f1C85618c5376ee4F38e43eE53b63': 2509787861801245,
        '0x726022a9fe1322fA9590FB244b8164936bB00489': 2.1798961394615612e+21,
        '0x6eF2376fA6e12Dabb3a3ED0Fb44E4ff29847Af68': 100420
      }
    ]

      at Object.<anonymous> (test/strategy.test.ts:74:15)

  console.log
    Resolved in 0.17 sec.

      at Object.<anonymous> (test/strategy.test.ts:75:15)

  console.log
    Scores with latest snapshot [
      {
        '0xeD2bcC3104Da5F5f8fA988D6e9fAFd74Ae62f319': 184000000000000000000,
        '0x3c4B8C52Ed4c29eE402D9c91FfAe1Db2BAdd228D': 7.469258545106344e+24,
        '0xd649bACfF66f1C85618c5376ee4F38e43eE53b63': 2509787861801245,
        '0x726022a9fe1322fA9590FB244b8164936bB00489': 2.1798961394615612e+21,
        '0x6eF2376fA6e12Dabb3a3ED0Fb44E4ff29847Af68': 100420
      }
    ]

      at Object.<anonymous> (test/strategy.test.ts:160:15)

  console.log
    Resolved in 0.07 sec.

      at Object.<anonymous> (test/strategy.test.ts:161:15)

 PASS  test/strategy.test.ts (22.628 s)

Test strategy "api" with example index 0
    √ Strategy name should be lowercase and should not contain any special char expect hyphen (3 ms)
    √ Strategy name should be same as in examples.json (1 ms)
    √ Addresses in example should be minimum 3 and maximum 20 (1 ms)
    √ Must use a snapshot block number in the past (396 ms)
    √ Strategy should run without any errors (224 ms)
    √ Should return an array of object with addresses (1 ms)
    √ Should take less than 10 sec. to resolve
    √ File examples.json should include at least 1 address with a positive score
    √ Returned addresses should be checksum addresses (2 ms)
    √ Voting power should not depend on other addresses (104 ms)

Test strategy "api" with example index 0 (latest snapshot)
    √ Strategy should run without any errors (576 ms)
    √ Should return an array of object with addresses (1 ms)

Test strategy "api" with example index 0 (with 500 addresses)
    ○ skipped Should work with 500 addresses
    ○ skipped Should take less than 20 sec. to resolve with 500 addresses

Other tests with example index 0
    ○ skipped Check schema (if available) is valid with examples.json
    ○ skipped Strategy should work even when strategy symbol is null

Others:
    √ Author in strategy should be a valid github username (287 ms)
    √ Version in strategy should be a valid string

Test Suites: 1 passed, 1 total
Tests:       4 skipped, 14 passed, 18 total
Snapshots:   0 total
Time:        23.633 s
Ran all test suites matching /strategy.test.ts/i.
```

**Review by ChaituVR — APPROVED** [review]
Thanks @ori-wagmi

**Pull request merged**